src/encoded/static/components/top_hits/hit.js

```diff
@@ -0,0 +1,174 @@
+
+Title.propTypes = { value: PropTypes.string.isRequired };
+
+Other.propTypes = { value: PropTypes.string.isRequired };
+
+
+/**
+* Maps a field to the formatting component used to style
+* that part of the hit.
+*/
+const fieldToComponent = {
+    name: Accession,
+    title: Title,
+    target: Other,
+    details: Other,
+    description: Description,
+};
+
+
+/**
+* Shortens a sentence to maxWords.
+*/
+export const maybeTrimByWord = (value, maxWords) => {
+    const splitValue = value.split(WHITESPACE);
+    if (splitValue.length > maxWords) {
+        return `${splitValue.slice(0, maxWords).join(WHITESPACE)}...`;
+    }
+    return value;
+};
+
+
+/**
+* Some fields are lists and should be converted to strings.
+*/
+export const maybeConvertArrayToString = (value, sep = ', ') => {
+    if (Array.isArray(value)) {
+        return value.join(sep);
+    }
+    return value;
+};
+
+
+/**
+* Encapsulates all the formatting for a raw top hit result. This
+* tries to generalize across most objects and return the first
+* value for that category that exists. It's important to remember
+* that these fields have to be specifically requested in the params
+* of the query URL (i.e. ?field=param) defined in ./constants.
+*/
+class Hit {
+    constructor(item) {
+        this.item = item;
+    }
+
+    formatTitle() {
+        return (
+            this.item.assay_title ||
+            this.item.assay_term_name ||
+            this.item.term_name
+        );
+    }
+
+    formatTarget() {
+        return this.item.target && this.item.target.title;
+    }
+
+    formatName() {
+        return this.item.accession || this.item['@id'];
+    }
+
+    formatDescription() {
+        return (
+            this.item.biosample_summary ||
+            this.item.summary ||
+            this.item.description ||
+            this.item.output_type ||
+            this.item.classification ||
+            this.item.antigen_description ||
+            this.item.title
+        );
+    }
+
+    formatDetails() {
+        return this.item.file_type;
+    }
+
+    /**
+    * This controls the order of values that are displayed in a
+    * result. The field string is used to recover the appropriate
+    * formatting component and also as a unique rendering key. The
+    * values are converted to a string (in case they are a list), and
+    * then empty values are filtered out (e.g. not every object
+    * has a target etc.).
+    */
+    getValues() {
+        return [
+            ['name', this.formatName()],
+            ['title', this.formatTitle()],
+            ['target', this.formatTarget()],
+            ['details', this.formatDetails()],
+            ['description', this.formatDescription()],
+        ].map(
+            ([field, value]) => (
+                [
+                    field,
+                    maybeConvertArrayToString(value),
+                ]
+            )
+        ).filter(
+            ({ 1: value }) => Boolean(value)
```

Comment:
Yeah, there was originally a third item in the list, and I was only trying to pull out the middle value. Couldn't seem to make eslint happy with array destructuring: 

`([, value, ]) => Boolean(value)` caused 
``` 
150:22  error  Unexpected trailing comma            comma-dangle
150:23  error  There should be no space before ']'  array-bracket-spacing
```

while removing the space `([, value,]) => Boolean(value)` caused:
```
150:22  error  Unexpected trailing comma      comma-dangle
150:22  error  A space is required after ','  comma-spacing
```
Found some suggestion about using this object destructuring syntax instead, but agree it's weird.

I think I can go back to `[, value]` now that there is only two values.

Reply:
Ah, I see. yeah, with a third value you’d get that. Can probably avoid it with an awkward:

```
[
    ,
    value,
    ,
]
```

I haven't actually checked if ESLint is fine with that. Maybe we should turn off the comma-dangle rule permanently, because it doesn't take this case into account, and it's a valid case. If you want to turn off the command-dangle rule in .eslintrc.js, I wouldn’t mind.

Reply:
Well, actually, if you had three values, does it complain if you just ignore the third by leaving out the trailing comma? I would think it just gets ignored, which is what you want anyway.

Reply:
Looks like ignoring works:

```
> a = [[1, 2, 3], [4, 5, 6], [7, 8, 9]];
[ [ 1, 2, 3 ], [ 4, 5, 6 ], [ 7, 8, 9 ] ]
> a.map(([,b]) => { console.log(b); })
2
5
8
```

Reply:
Ah, yeah! I think that's the right way. Good to know.